### DISCUSSION TITLE
Fluent Migrator and embedded sql resources

### DISCUSSION THREAD

**Discussion by erehtolleh** [serenity-is/Serenity#5377] [Help] [answered]
I am using Serenity 5.0.26. .Net 5

I had an issue with using Fluent Migrator recently where I had created a bunch of sql statements files to be executed as an embedded resources.  I had an error of it not being able to find the resources and was able to fix it with the steps below.

In order to do this, you have to:
1. Create the sql file with your sql statements (*.sql)
      The important thing here is to go to property of file and select Embeded Resource for Build Action, Do not copy for copy to output directory.
2. Create a migration as you would for Fluent Migrator
             public override void Up()
        {
            Execute.EmbeddedScript("nameofyourfile.sql"); 
        }
3. Add this line to DataMigration.cs in RunMigrations()
                     builder.WithGlobalConnectionString(cs.ConnectionString);
                    builder.WithMigrationsIn(typeof(DataMigrations).Assembly);
                    builder.ScanIn(typeof(DataMigrations).Assembly).For.EmbeddedResources(); //add this line

I got help from this [posted issue](https://github.com/fluentmigrator/fluentmigrator/issues/730).

I think you can change the path via step two to avoid changes in step 3 which I would have preferred so as not to touch the default Serenity file.  I tried different paths but was unsuccessful.  The steps above was successful for me.

**Comment by minhhungit** [comment, marked as answer]
Thanks for sharing  @erehtolleh , I aslo created a tool to work with migration based on fluentmigrator for myself

It is still good when Serenity embed migration to website, it will reduce complex for new developer, but I want to spit database tool and website

If you are interested in my tool or if someone want to move migration out of website, try this https://github.com/minhhungit/DatabaseMigrateExt
